### PR TITLE
(CDAP-4687) Added system metadata for the artifact created for the ap…

### PR DESCRIPTION
**Pull request by bdmogal**
…p when it is deployed without deploying the artifact first.

Jira: [CDAP-4687](https://issues.cask.co/browse/CDAP-4687)
Build: http://builds.cask.co/browse/CDAP-RBT613

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java [hunk 1]
```diff
@@ -305,6 +306,9 @@ public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile)
       ArtifactClasses artifactClasses = inspectArtifact(artifactId, artifactFile, null, parentClassLoader);
       validatePluginSet(artifactClasses.getPlugins());
       ArtifactMeta meta = new ArtifactMeta(artifactClasses, ImmutableSet.<ArtifactRange>of());
+      ArtifactInfo artifactIfo = new ArtifactInfo(artifactId.toArtifactId(), artifactClasses,
```
Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java]: nit  -  misspelled variable name

**Comment by anwar6953** [comment]
Remainder LGTM

**Comment by bdmogal** [comment]
Fixed variable name. Will merge after successful build.

**Comment by bdmogal** [comment]
Build passed - http://builds.cask.co/browse/CDAP-RBT613-1. Merging.

**Pull request merged**